src/main/scala/ai/privado/languageEngine/javascript/passes/read/GraphqlQueryParserPass.scala

```diff
@@ -0,0 +1,208 @@
+        }
+
+      case Failure(error) =>
+        logger.info(s"Syntax error: ${error.getMessage}")
+    }
+  }
+
+  // Function to extract fields from ObjectTypeDefinition
+  def extractObjectTypeDetails(builder: DiffGraphBuilder, node: Expression, objDef: ObjectTypeDefinition): Unit = {
+    val typeName = objDef.name
+    val fields = objDef.fields.flatMap {
+      case field: FieldDefinition =>
+        val fieldName      = field.name
+        val fieldType      = field.fieldType
+        val inputValueDefs = extractInputValueDefinitions(field.arguments.toList)
+        s"Field: $typeName.$fieldName" :: inputValueDefs.map(s => s"$typeName.$fieldName.$s")
+      case null => Nil
+    }
+
+    processGQLReadNode(builder, node, typeName, fields.toList)
+  }
+
+  // Function to extract fields
+  def extractFields(selections: List[Selection]): List[String] = {
+    selections.flatMap {
+      case field: Field =>
+        val nestedFields = extractFields(field.selections.toList)
+        // field.name :: nestedFields.map(nestedField => s"${field.name}.$nestedField")
+        field.name :: nestedFields.map(nestedField => nestedField)
+      case _ => Nil
+    }
+  }
+
+  // Function to extract variable definitions
+  def extractVariableDefinitions(variableDefinitions: List[VariableDefinition]): List[String] = {
+    variableDefinitions.map(definition => definition.name)
+  }
+
+  // Function to extract object type definitions
+  def extractObjectTypeDefinitions(definitions: List[ObjectTypeDefinition]): List[String] = {
+    definitions.map(definition => definition.name)
+  }
+
+  // Function to extract input value definitions
+  def extractInputValueDefinitions(definitions: List[InputValueDefinition]): List[String] = {
+    definitions.map(definition => definition.name)
+  }
+
+  def processGQLReadNode(builder: DiffGraphBuilder, node: Expression, tableName: String, columns: List[String]) = {
+    // Match classes which end with tableName/ObjectName
+    val sensitiveMemberRuleIds = sensitiveClasses.find(s => s.matches(s"(?i).*${tableName}")) match {
+      case Some(value) => sensitiveClassesWithMatchedRules(value).keys.l
+      case None        => List.empty
+    }
+
+    if (columns.length == 1 && columns(0) == "*") {
+      if (sensitiveMemberRuleIds.nonEmpty)
+        sensitiveMemberRuleIds.foreach(ruleId => addTagsToNode(ruleId, node, builder))
+      else {
+        /* Run dataflow and verify the data-elements read from the call,
+              Ex - resultSet = statement.executeQuery("SELECT * FROM mytable");
+              // Loop through the result set and print out each row
+              while (resultSet.next()) {
+                  int id = resultSet.getInt("id");
+                  String firstName = resultSet.getString("name");
+                  int age = resultSet.getInt("age");
+                  System.out.println("ID: " + id + ", Name: " + firstName + ", Age: " + age)
+              }
+         */
+        val dataElementSinks =
```

Comment:
The code present is else block is very expensive, we have placed it already under a flag in database read pass, and intend to optimized or remove that. Don't use this code